client/lib/plugins/store.js

```diff
@@ -248,8 +243,8 @@ PluginsStore = {
 		if ( ! plugins ) {
 			return;
 		}
-		plugins = plugins.filter( _filters.isEqual.bind( this, pluginSlug ) );
-		plugin = plugins.pop();
+
+		plugin = find( plugins, _filters.isEqual.bind( this, pluginSlug ) );
```

Comment:
Well ideally it would be sorted by plugin version number. Since different sites could be running different versions of the plugin. But I don't think we get much info since more information gets pulled from wp.org if there is any available. I think for right now the order doesn't really matter.